packages/chart-advisor/src/advisor/index.ts

```diff
@@ -147,12 +155,12 @@ export class Advisor {
     const { data, dataProps, smartColor, options, colorOptions } = params;
     // otherwise the input data will be mutated
     const copyData = cloneDeep(data);
-    const { fields } = params as ChartAdviseParams;
+    const { fields, fieldTypes } = params as ChartAdviseParams;
     // transform data into DataFrame
     let dataFrame: DataFrame;
     try {
       if (fields) {
-        dataFrame = new DataFrame(copyData, { columns: fields });
+        dataFrame = new DataFrame(copyData, { columns: fields, columnTypes: fieldTypes });
```

Comment:
Plz note that DW will directly convert the data type, not just modify the info.

Reply:
Yes，it is used only in the case that the data is not formated to correct type， for example data is all string from backend
If we don't support this, I need to convert the data type by another new DataFrame which will take wasted effort. Maybe we can add comment to tell user about this

Reply:
got it, we will also add more information in v3 docs.